### PR TITLE
`managed_hsm_*`: add managed hsm nested key ids

### PR DESCRIPTION
**Pull request by wuxu92**
Dependent on https://github.com/hashicorp/go-azure-helpers/pull/222.

## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description

add the managed hsm key id which is required by #25069 and other places.

## PR Checklist

- [ ] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [ ] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [ ] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [ ] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [ ] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [ ] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [ ] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [ ] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [ ] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_resource` - support for the `thing1` property [GH-00000]

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/managedhsm/resourceids.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package managedhsm
+
+//go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=ManagedHSMNestedKeyWithVersion -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.KeyVault/managedHSMs/mhsm1/keys/key1/versions/version1
+//go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=ManagedHSMNestedKeyVersionless -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.KeyVault/managedHSMs/mhsm1/keys/key1
```
Comment (tombuildsstuff) [inline internal/services/managedhsm/resourceids.go:7]: The legacy Resource ID generator shouldn't be used for new Resources going forwards - can we instead update these to use [the new Resource ID interface](https://github.com/hashicorp/go-azure-helpers/blob/891641b735c70484dffd36a0fba21bca79efc603/resourcemanager/resourceids/interface.go#L6-L20) that `hashicorp/go-azure-sdk` etc uses?

Comment (tombuildsstuff) [reply internal/services/managedhsm/resourceids.go:7]: Also shouldn't these be the Data Plane IDs rather than the Resource Manager IDs?

**Comment by rcskosir** [comment]
@wuxu92 👋 Thanks for using our new PR template and including a description! Can you please update it to include an accurate changelog portion and check the appropriate boxes for this PR? This helps with the review process and also future users understanding of what is happening in this PR. Thanks!

**Comment by katbyte** [comment]
@wuxu92 we have a ton of CI failures?

**Comment by tombuildsstuff** [comment]
hey @wuxu92 

Thanks for this PR - taking a look through here there's already an existing Resource Manager ID parser within `hashicorp/go-azure-sdk` (which we should probably promote to be a Common ID - I've opened https://github.com/hashicorp/go-azure-helpers/issues/230 to track that for the future) - and as such this wants to focus on the Data Plane component instead.

I've ended up digging into this and have discovered that we want to take a different approach than the one used here, since this'll also end up being used for Key Vault in the future too - and as such I hope you don't mind but I'm going to close this PR in favour of https://github.com/hashicorp/terraform-provider-azurerm/pull/25601 which implements the Data Plane specific Resource ID parsers.

Thanks!

**Comment by katbyte** [comment]
superceeded by https://github.com/hashicorp/terraform-provider-azurerm/pull/25601

**Pull request closed without merging**